src/dawn/CodeGen/CodeGen.cpp

```diff
@@ -27,6 +28,208 @@ size_t CodeGen::getVerticalTmpHaloSizeForMultipleStencils(
              : 0;
 }
 
+std::string CodeGen::generateGlobals(std::shared_ptr<SIR> const& sir,
+                                     std::string namespace_) const {
+
+  const auto& globalsMap = *(sir->GlobalVariableMap);
+  if(globalsMap.empty())
+    return "";
+
+  std::stringstream ss;
+
+  Namespace cudaNamespace(namespace_, ss);
+
+  std::string StructName = "globals";
+
+  Struct GlobalsStruct(StructName, ss);
+
+  for(const auto& globalsPair : globalsMap) {
+    sir::Value& value = *globalsPair.second;
+    std::string Name = globalsPair.first;
+    std::string Type = sir::Value::typeToString(value.getType());
+
+    GlobalsStruct.addMember(Type, Name);
+  }
+  auto ctr = GlobalsStruct.addConstructor();
+  for(const auto& globalsPair : globalsMap) {
```

Comment:
I think would be weird, because first we fill all the Members, and then we need to create the constructor GlobalsStruct.addConstructor (out of the loop)

Reply:
ok, good argument